vassal-app/src/main/java/VASSAL/build/module/map/deck/DeckSortKeyCommand.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *
+ * Copyright (c) 2021 by The VASSAL Development Team
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License (LGPL) as published by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.build.module.map.deck;
+
+import VASSAL.build.AutoConfigurable;
+import VASSAL.build.GameModule;
+import VASSAL.configure.Configurer;
+import VASSAL.configure.ConfigurerFactory;
+import VASSAL.counters.Deck;
+import VASSAL.counters.KeyCommand;
+import VASSAL.i18n.Resources;
+import VASSAL.tools.NamedKeyStroke;
+import VASSAL.tools.NamedKeyStrokeListener;
+
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+public class DeckSortKeyCommand extends AbstractDeckKeyCommand {
+
+  public static final String SORT_PARAMETERS = "sortParameters";
+
+  private List<SortParameter> sortParameters;
+  private NamedKeyStrokeListener sortListener;
+
+  public static String getConfigureTypeName() {
+    return Resources.getString("Editor.DeckSortKeyCommand.component_type"); //$NON-NLS-1$
+  }
+
+  public List<SortParameter> getSortParameters() {
+    if (sortParameters == null) {
+      // Start with a default entry
+      sortParameters = new ArrayList<>();
```

Comment:
I saw this and thought that surely `ArrayList` had a variadic constructor and this could be done in one line... but it doesn't! Lame.